images/base/Dockerfile

```diff
@@ -112,7 +113,9 @@ RUN echo "Ensuring scripts are executable ..." \
  && echo "Adjusting systemd-tmpfiles timer" \
     && sed -i /usr/lib/systemd/system/systemd-tmpfiles-clean.timer -e 's#OnBootSec=.*#OnBootSec=1min#' \
  && echo "Modifying /etc/nsswitch.conf to prefer hosts" \
-    && sed -i /etc/nsswitch.conf -re 's#^(hosts:\s*).*#\1dns files#'
+    && sed -i /etc/nsswitch.conf -re 's#^(hosts:\s*).*#\1dns files#' \
+ && echo "Enabling DNS caching" \
+    && systemctl enable nscd
```

Comment:
i'm not so sure this is the way to go about this
https://unix.stackexchange.com/a/6240

Reply:
basically we only want to cache the calls to the node names `kind-control-plane`, those are hostnames are not going to change and we don't care about its TTL (if they change is because the container was restarted and the cache is empty).

nscd intercepts the libc calls (gethostby*), in this case if we can avoid things like kubelet asking for the node ip periodically (every 60 sec???) to get the node IP.

let me double check how nscd interact with docker and pods and kubernetes 🤔 , it's important that does not intercepts the pods querys to coredns and coredns to external

Reply:
is the cache actually empty? /var is persistent.

Reply:
don't know the internals, but I think that /var/run/nscd is only needed to create the socket